### PR TITLE
chore: add lint

### PR DESCRIPTION
**Pull request by neo773**
/claim  #505

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -98,6 +98,15 @@ jobs:
...
           args: --all-targets --all-features -- -D warnings
 
+  lint:
+    name: Run Lint
+    if: contains(github.event.pull_request.labels.*.name, 'ci:lint')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run lint script
+        run: ./lint.sh
```
Comment (tusharmath) [inline .github/workflows/ci.yml:108]: It needs to commit and push the changes.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -98,6 +98,21 @@ jobs:
           command: clippy
           args: --all-targets --all-features -- -D warnings
 
+  lint:
+    name: Run Lint
```
Comment (tusharmath) [inline .github/workflows/ci.yml:102]:
```suggestion
    name: CI:Lint
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -98,6 +98,21 @@ jobs:
...
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run lint script
+        run: ./lint.sh
+      - name: Commit and push if changed
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: Apply lint changes
```
Comment (tusharmath) [inline .github/workflows/ci.yml:114]:
```suggestion
          commit_message: style: apply lint fixes
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -98,6 +98,21 @@ jobs:
...
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Run lint script
+        run: ./lint.sh
+      - name: Commit and push if changed
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: style: apply lint fixes
```
Comment (tusharmath) [inline .github/workflows/ci.yml:114]:
```suggestion
          commit_message: style: apply lint fixes
          
          
          
          
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -98,6 +98,25 @@ jobs:
...
+          
+          
+          
+          
+
```
Comment (tusharmath) [inline .github/workflows/ci.yml:119]:
```suggestion
          commit_message: style: apply lint fixes
```

**Comment by tusharmath** [comment]
Not sure why the build is not running @neo773

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -98,6 +98,20 @@ jobs:
           command: clippy
           args: --all-targets --all-features -- -D warnings
 
+  lint:
+    name: CI:Lint
+    if: contains(github.event.pull_request.labels.*.name, 'ci:lint')
+    runs-on: ubuntu-latest
```
Comment (tusharmath) [inline .github/workflows/ci.yml:104]:
```suggestion
    runs-on: ubuntu-latest  
```

**Comment by tusharmath** [comment]
Figured out why the build wasn't running. The YML was invalid. Fixed that. There is another issue, the build doesn't get triggered automatically if I add the label.

**Comment by neo773** [comment]
@tusharmath Hey, sorry about that was afk so couldn't reply

**Comment by tusharmath** [comment]
closing in favour of #507

**Pull request closed without merging**